backend/src/main/java/gov/cdc/usds/simplereport/config/AzureTestEventReportingQueueConfiguration.java

```diff
@@ -22,24 +24,45 @@
 @Slf4j
 @Configuration
 class AzureTestEventReportingQueueConfiguration {
-  @Bean
-  @Primary
+  @Bean("csvQueueReportingService")
   @ConditionalOnProperty(
       value = "simple-report.azure-reporting-queue.enabled",
       havingValue = "true")
   TestEventReportingService storageQueueReportingService(
-      ObjectMapper mapper, QueueAsyncClient queueClient) {
+      ObjectMapper mapper, @Qualifier("csvQueue") QueueAsyncClient queueClient) {
     log.info("Configured for queue={}", queueClient.getQueueName());
     return new AzureStorageQueueTestEventReportingService(mapper, queueClient);
   }
 
+  @Bean("fhirQueueReportingService")
+  @ConditionalOnProperty(
+      value = "simple-report.azure-reporting-queue.fhir-queue-enabled",
+      havingValue = "true")
+  TestEventReportingService fhirQueueReportingService(
+      FhirContext context, @Qualifier("fhirQueueClient") QueueAsyncClient queueClient) {
+    log.info("Configured for queue={}", queueClient.getQueueName());
+    return new AzureStorageQueueFhirReportingService(context, queueClient);
+  }
+
   @Bean
-  @ConditionalOnMissingBean
-  TestEventReportingService noOpReportingService() {
+  FhirContext fhirContext() {
+    return FhirContext.forR4();
+  }
+
+  @Primary
+  @Bean(name = "csvQueueReportingService")
+  @ConditionalOnMissingBean(name = "csvQueueReportingService")
+  TestEventReportingService noOpCSVReportingService() {
     return new NoOpReportingService();
   }
 
-  @Bean
+  @Bean(name = "fhirQueueReportingService")
+  @ConditionalOnMissingBean(name = "fhirQueueReportingService")
+  TestEventReportingService noOpFhirReportingService() {
+    return new NoOpReportingService();
```

Comment:
I think that using the `NoOpReportingService` for FHIR plus having `fhir-queue-enabled` disabled in prod will result in PII in our logs. The `NoOpReportingService` tries to convert a `TestEvent` to a `TestEventExport` and logs the outcome, which we don't want in production. 

It is handy to have  the existing`NoOpReportingService` locally though, because it allows us to quickly catch serialization issues with `TestEventExport`. My suggestion would be to have a FHIR-specific `noOpReportingService` that just logs when the service is called but doesn't log anything about the actual `TestEvent`.

Reply:
Makes sense, I put a profile on the bean for non prod only since I don't think we should ever debug like that there.